L1Trigger/L1THGCal/plugins/ntuples/HGCalTriggerNtupleHGCDigis.cc

```diff
@@ -33,16 +34,17 @@ class HGCalTriggerNtupleHGCDigis : public HGCalTriggerNtupleBase
         std::vector<int> hgcdigi_wafer_;
         std::vector<int> hgcdigi_wafertype_ ;
         std::vector<int> hgcdigi_cell_;
-        std::vector<float> hgcdigi_cell_eta_;
-        std::vector<float> hgcdigi_cell_phi_;
-        std::vector<float> hgcdigi_cell_z_;
+        std::vector<float> hgcdigi_eta_;
+        std::vector<float> hgcdigi_phi_;
+        std::vector<float> hgcdigi_z_;
         std::vector<uint32_t> hgcdigi_data_;
         std::vector<int> hgcdigi_isadc_;
-        std::vector<int> hgcdigi_istot_;
-        std::vector<float> hgcdigi_cell_simhit_energy_;
+        std::vector<float> hgcdigi_simenergy_;
 
         edm::ESHandle<HGCalGeometry> geom_ee, geom_fh;
         edm::ESHandle<HGCalTopology> topo_ee, topo_fh;
+        std::unordered_map<uint32_t, double> simhits_ee_;
+        std::unordered_map<uint32_t, double> simhits_fh_;
```

Comment:
You need to clean these two maps at each event. Or you can also instead define them locally in the fill() function and pass them as arguments of the simhits() function (this way they are destroyed automatically at each event).